### ISSUE TITLE
Looking for a way to not exit on async.parallel on error.

### ISSUE THREAD

**Issue by reggi** [caolan/async#675]
Majority of my most recent use-cases involve async functions that aren't dependent on each other, and so when some async function breaks up top, the rest of my code doesn't execute. Wondering if there's a simple way to do something like this without being so obtrusive to the library / so much overhead. In this case I'd like to return false on error, so I'm building an object with every property intact, if something fails everything is else is still there, and false takes the broken async functions place.

```
var noError = function(func) {
  return function(callback) {
    func(function(err, value) {
      if (err) return callback(null, false);
      return callback(null, value);
    });
  }
}

var parallelNoError = function(functions, callback) {
  var newFunctions = {};
  for (var func in functions) {
    newFunctions[func] = noError(functions[func]);
  }
  return async.parallel(newFunctions, callback);
}
```

@aearly Would love your thoughts on this one too.

**Comment by goloroden** [comment]
+1, would love to see an option to continue on errors, too (although I don't need it for `parallel`, but for `forEach`).

**Comment by danielshir** [comment]
+1 for continuing on errors.
I have encountered a case where some functions running in parallel() will require cleanup code to run after completion. The current design scheme makes it hard to do so if an error occurs.

**Comment by colinmegill** [comment]
+1

**Comment by oillescas** [comment]
+1

**Comment by skilledDeveloper** [comment]
+1

**Comment by fizerkhan** [comment]
+1

**Comment by Levino** [comment]
+1

**Comment by jvillanti** [comment]
+1

**Comment by aearly** [comment]
Closing this in favor of #942 . There's also the PR: #1012

**Comment by chandlerenterprises** [comment]
+1
